hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/invalidation/InvalidationMetaDataFetcher.java

```diff
@@ -115,18 +115,23 @@ private void processMemberMetadata(Member member, InternalCompletableFuture futu
         MetadataHolder resultHolder = new MetadataHolder();
         try {
             extractMemberMetadata(member, future, resultHolder);
-        } catch (HazelcastInstanceNotActiveException e) {
-            logger.finest(e);
         } catch (Exception e) {
-            if (logger.isWarningEnabled()) {
-                logger.warning(format("Can't extract invalidation meta-data of %s", member), e);
-            }
+            handleExceptionWhileProcessingMetadata(member, e);
+            return;
         }
 
         repairUuids(resultHolder.partitionUuidList, handlers);
         repairSequences(resultHolder.namePartitionSequenceList, handlers);
     }
 
+    protected void handleExceptionWhileProcessingMetadata(Member member, Exception e) {
+        if (e instanceof HazelcastInstanceNotActiveException) {
```

Comment:
what about handling only `IllegalStateException` here? By doing this we can handle all possible cases i think, for example `HazelcastClientNotActiveException` is missing now despite the fix in this PR.